dotCMS/src/main/java/com/dotcms/rest/CMSConfigResource.java

```diff
@@ -497,4 +500,30 @@ public Response deleteEndpoint ( @Context HttpServletRequest request,
         return responseResource.response( responseMessage.toString() );
     }
 
+    @POST
+    @Path ("/regenerateKey")
+    @Produces (MediaType.APPLICATION_JSON)
+    @Consumes (MediaType.APPLICATION_FORM_URLENCODED)
+    public Response regenerateKey ( @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response) throws JSONException, IOException {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            final CompanyAPI companyAPI = APILocator.getCompanyAPI();
+            final Company defaultCompany = companyAPI.getDefaultCompany();
+            final Company updatedCompany = APILocator.getCompanyAPI().regenerateKey(defaultCompany, user);
```

Comment:
Use the `companyAPI` object you just created

Reply:
Good catch.